spark/src/main/scala/org/apache/spark/sql/delta/TableFeature.scala

```diff
@@ -520,6 +521,13 @@ object IcebergCompatV1TableFeature extends WriterFeature(name = "icebergCompatV1
   override def requiredFeatures: Set[TableFeature] = Set(ColumnMappingTableFeature)
 }
 
+/**
+ * Clustering table feature is enabled when a table is created with CLUSTER BY clause.
+ */
+object ClusteringTableFeature extends WriterFeature("clustering") {
+  override val requiredFeatures: Set[TableFeature] = Set(DomainMetadataTableFeature)
```

Comment:
Nit: maybe add a comment explaining what domain metadata is used for

Reply:
Thanks, I will address this in the next PR since the delta spec on the format of columns' names stored are not finalized.